src/main/java/com/github/timm/cucumber/generate/CucumberITGenerator.java

```diff
@@ -193,10 +199,13 @@ private String createFormatStrings() {
 
         for (int i = 0; i < formatStrs.length; i++) {
             final String formatStr = formatStrs[i].trim();
-            sb.append(String.format("\"%s:%s/%s.%s\"", formatStr,
-                    config.getCucumberOutputDir()
-                    .replace('\\', '/'), fileCounter, formatStr));
-
+            sb.append(String.format("\"%s:%s/%s/%s.%s\"", formatStr,config.getCucumberOutputDir().replace('\\', '/'),FilenameUtils.removeExtension(outputFileName), FilenameUtils.removeExtension(outputFileName), formatStr));
+            if(formatStr.equalsIgnoreCase("html"))
+                htmlFormat =String.format("\"%s:%s/%s/%s.%s\"", formatStr,config.getCucumberOutputDir().replace('\\', '/'),FilenameUtils.removeExtension(outputFileName), FilenameUtils.removeExtension(outputFileName), formatStr);
+            if(formatStr.equalsIgnoreCase("json"))
+                jsonFormat =String.format("\"%s:%s/%s/%s.%s\"", formatStr,config.getCucumberOutputDir().replace('\\', '/'),FilenameUtils.removeExtension(outputFileName), FilenameUtils.removeExtension(outputFileName), formatStr);
+            if(formatStr.equalsIgnoreCase("rerun"))
+                rerunFormat =String.format("\"%s:%s/%s/%s.%s\"", formatStr,config.getCucumberOutputDir().replace('\\', '/'),FilenameUtils.removeExtension(outputFileName), FilenameUtils.removeExtension(outputFileName), formatStr);
```

Comment:
I agree with @JasonSmiley208: if there is a standard extension, we should use that, to minimise interoperability issues with other plugins, etc.

* The rerun file extension is user specified (i.e. this plugin), so not worried about that
* The junit extension is .xml as "standard"